net/procfs/net_tcp.c

```diff
@@ -0,0 +1,211 @@
+
+      if (buflen - len < TCP_LINELEN)
+        {
+          break;
+        }
+
+#if defined(CONFIG_NET_IPv4)
+      if (domain == PF_INET)
+        {
+          laddr = &conn->u.ipv4.laddr;
+          raddr = &conn->u.ipv4.raddr;
+        }
+#endif /* CONFIG_NET_IPv4 */
+
+#if defined(CONFIG_NET_IPv6)
+      if (domain == PF_INET6)
+        {
+          laddr = &conn->u.ipv6.laddr;
+          raddr = &conn->u.ipv6.raddr;
+        }
+#endif /* CONFIG_NET_IPv6 */
+
+      len += snprintf(buffer + len, buflen - len,
+                      "    %2" PRIu8
+                      ": %02" PRIx8
+                      " %3" PRIx8 " %3" PRIu8
+                      " %3" PRIu8
+#if defined(CONFIG_NET_TCP_WRITE_BUFFERS) || \
+    defined(CONFIG_NET_TCP_WINDOW_SCALE)
+                      " %4" PRIu32
+#else
+                      " %4" PRIu16
+#endif
+                      " %3" PRIu8
+#if CONFIG_NET_SEND_BUFSIZE > 0
+                      " %6" PRIu32
+#endif
+                      " %6u",
+                      priv->offset++,
+                      conn->tcpstateflags,
+                      conn->sconn.s_flags,
+                      conn->crefs,
+                      conn->timer,
+                      conn->tx_unacked,
+                      conn->nrtx,
+#if CONFIG_NET_SEND_BUFSIZE > 0
+                      tcp_wrbuffer_inqueue_size(conn),
+#endif
+                      (conn->readahead) ? conn->readahead->io_pktlen : 0);
+
+      len += snprintf(buffer + len, buflen - len,
+                      " %*s:%-6" PRIu16 " %*s:%-6" PRIu16 "\n",
+                      (domain == PF_INET6) ? addrlen / 2 : addrlen,
+                      inet_ntop(domain, laddr, local, addrlen),
+                      ntohs(conn->lport),
+                      (domain == PF_INET6) ? addrlen / 2 : addrlen,
+                      inet_ntop(domain, raddr, remote, addrlen),
+                      ntohs(conn->rport));
+    }
+
+  net_unlock();
+
+  return len;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: netprocfs_read_tcpstats
+ *
+ * Description:
+ *   Read and format TCP statistics.
+ *
+ * Input Parameters:
+ *   priv - A reference to the network procfs file structure
+ *   buffer - The user-provided buffer into which network status will be
+ *            returned.
+ *   bulen  - The size in bytes of the user provided buffer.
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success; a negated errno value is returned
+ *   on failure.
+ *
+ ****************************************************************************/
+
+ssize_t netprocfs_read_tcpstats(FAR struct netprocfs_file_s *priv,
+                                FAR char *buffer, size_t buflen)
+{
+  int skip = 1;
+  int len = 0;
+
+  net_lock();
+
+  if (tcp_nextconn(NULL) != NULL)
```

Comment:
why check this?

Reply:
to avoid print title if there is no any tcp connections